### ISSUE TITLE
JSON schema: unexpected error message

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/robotframework-testsuitesmanagement#226]
The JSON code

```
"params" : {
   "global" : {
               "teststring" : "eins",
               "${params.global.teststring}" : "${params.global.teststring}.zwei"
   }
}
```

causes the following error message:

```
Verification against JSON schema failed: 'Additional properties are not allowed ('eins' was unexpected)'
Please put the additional params into 'params': { 'global': {...}
```

But expected is value `"eins.zwei"`.

**Comment by HolQue** [comment]
Addendum:

This works:

`${params.global.teststring} : "${params.global.teststring}.zwei"`

But with quotes on the left hand side of the colon

`"${params.global.teststring}" : "${params.global.teststring}.zwei"`

the meaning is: create a new parameter with name `"eins"`.

Internally, after dollar operator expressions are substituted, the code should be 

"eins" : "eins.zwei"

And it seems to me that while replacing the dollar operator expressions, the scope gets lost for new parameters. This causes the schema error. And this happens internally. Because when I place this line

"eins" : "eins.zwei"

hard coded in the JSON file, the parameter "eins" is created properly.

**Comment by namsonx** [comment]
Hello Holger,

You're right! I had checked and the scope of new added parameters are duplicated at the highest level of json configuration file. That is the reason of JSON schema check failed.
I created the new [commit 7c2eb63c79](https://github.com/test-fullautomation/python-jsonpreprocessor/commit/7c2eb63c79175aa940b7003e9fbc206e98bfa0ad) on stabi branch of JsonPreprocessor repository to fix this issue.

Thank you,
Son
